pagopa/io-app
# feat(Sicilia Vola): [IASV-31] Select flights date screen

## Short description
This PR completes the `SelectFlightsDateScreen` with the proposed UX.

This PR also introduces a new UI component `DateTimePicker`. 
It built a wrapper around the [react-native-modal-datetime-picker](https://github.com/mmazzarolo/react-native-modal-datetime-picker ) and allows to set the following properties:
- date
- onConfirm
- label
- minimumDate
- blocked

## List of changes proposed in this pull request
- Added the `DateTimePicker` component
- Added the `SelectFlightsDateScreen`

**IOS**

https://user-images.githubusercontent.com/11773070/128365545-25d6833f-5b25-4d69-ac1a-7f9026be08a5.mp4

**ANDROID**

https://user-images.githubusercontent.com/11773070/128365582-9e2c607b-a5b4-464f-ad8a-4f2a81f3b257.mov

## How to test
Modify in the `CheckStatusRouterScreen` the following lines:
https://github.com/pagopa/io-app/blob/0150ad3c78a7b842f02d929ab88ca5b43ce0bf23/ts/features/bonus/siciliaVola/screens/voucherGeneration/CheckStatusRouterScreen.tsx#L99-L100
in:
```
isServiceAlive: remoteReady(true),
tosAccepted: remoteReady(true)
```
Starts the flow from the SiciliaVola service detail and select in:
- `CheckStatusRouterScreen` -> `Resident in Sicily`
- `SelectBeneficiaryCategoryScreen` -> `Disabled`
- `DisabledAdditionalInfoScreen` -> continue

Note that when you press `Continue` also the Store ( `bonus/sv/voucherGeneration/voucherRequest/departureDate` and  `bonus/sv/voucherGeneration/voucherRequest/returnDate`) will be update based on your selection: `One way flight` or `Round trip`.

Make sure you have the updated version of the master branch on dev-server in order to see the service details.
